synapse/app/homeserver.py

```diff
@@ -185,6 +185,7 @@ def _listener_http(self, config, listener_config):
                         "/_matrix/client/r0": client_resource,
                         "/_matrix/client/unstable": client_resource,
                         "/_matrix/client/v2_alpha": client_resource,
+                        "/_matrix/versions": client_resource,
```

Comment:
this needs a change.

Having this list is *horrible*, though I realise fixing it is probably out of scope right now.